### PR TITLE
Add csv file reading support

### LINKED ISSUE #52: Implement `CsvfileReader` to `csvfile.py` adapter
Currently there is only writing support for CSV files, but in some cases it would be nice to also read CSV files and output them as records.

This does not have to be a fully featured unpacker that supports field types.

### PR DESCRIPTION
**Pull request by yunzheng**
Fixes #52

### CODE REVIEW

File: flow/record/adapter/csvfile.py [hunk 1]
```diff
@@ -52,3 +54,28 @@ def close(self):
...
         self.fp = None
+
+
+class CsvfileReader(AbstractReader):
+    fp = None
+
+    def __init__(self, path, selector=None, **kwargs):
+        self.selector = make_selector(selector)
+        if path in (None, "", "-"):
+            self.fp = sys.stdout
```
Comment (Schamper) [inline flow/record/adapter/csvfile.py:65]:
```suggestion
            self.fp = sys.stdin
```
I assume?

File: flow/record/adapter/csvfile.py [hunk 2]
```diff
@@ -52,3 +54,28 @@ def close(self):
...
+        self.reader = csv.reader(self.fp)
+
+    def close(self):
+        if self.fp:
+            self.fp.close()
+        self.fp = None
+
+    def __iter__(self):
+        row = next(self.reader)
+        desc = RecordDescriptor("csv/reader", [("string", col) for col in row])
```
Comment (Schamper) [inline flow/record/adapter/csvfile.py:77]: Maybe nice feature to support headerless CSV files and specify the columns through an argument of the adapter?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: flow/record/adapter/csvfile.py (hunk 1 above)
Comment (yunzheng) [reply flow/record/adapter/csvfile.py:65]: oh oops, nice catch.

File: flow/record/adapter/csvfile.py (hunk 2 above)
Comment (yunzheng) [reply flow/record/adapter/csvfile.py]: hmm yes, was thinking this as well. Will look into it.

Comment (yunzheng) [reply flow/record/adapter/csvfile.py]: Adding it to the adapter should be no problem, would you also expect this to be easy to do via `rdump` as well, for example via a new CLI argument, or would specifying a `csvfile://stuff.csv?fields=a,b,c` be sufficient?

Comment (Schamper) [reply flow/record/adapter/csvfile.py]: Yeah I meant the latter method, supplying through the adapter query arguments should be sufficient.

**Review by Schamper — APPROVED** [review]

**Pull request merged**